src/gluonts/model/n_beats/_network.py

```diff
@@ -0,0 +1,521 @@
+        seasonal_error = factor * F.mean(
+            F.abs(
+                F.slice_axis(whole_target, axis=1, begin=periodicity, end=None)
+                - F.slice_axis(whole_target, axis=1, begin=0, end=-periodicity)
+            ),
+            axis=1,
+        )
+        flag = seasonal_error == 0
+
+        mase = (
+            F.mean(F.abs(future_target - forecast), axis=1) * (1 - flag)
+        ) / (seasonal_error + flag)
+
+        return mase
+
+
+class NBEATSTrainingNetwork(NBEATSNetwork):
+    @validated()
+    def __init__(self, loss_function: str, freq: str, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.loss_function = loss_function
+        self.freq = freq
+
+        # Covert frequency string like "2H" to whats called the periodicity m.
+        # E.g. 12 in case of "2H" because of 12 times two hours in a day.
+        self.periodicity = get_seasonality(self.freq)
+
+    # noinspection PyMethodOverriding,PyPep8Naming
+    def hybrid_forward(
+        self, F, past_target: Tensor, future_target: Tensor
+    ) -> Tensor:
+        """
+
+        Parameters
+        ----------
+        F
+        past_target
+            Tensor with past observations.
+            Shape: (batch_size, context_length, target_dim).
+        future_target
+            Tensor with future observations.
+            Shape: (batch_size, prediction_length, target_dim).
+
+        Returns
+        -------
+        Tensor
+            Loss tensor. Shape: (batch_size, ).
+        """
+        # future_target never used
+        forecast = super().hybrid_forward(
+            F, past_target=past_target, future_target=future_target
+        )
+
+        if self.loss_function == "sMAPE":
+            loss = self.smape_loss(F, forecast, future_target)
+        elif self.loss_function == "MAPE":
+            loss = self.mape_loss(F, forecast, future_target)
+        elif self.loss_function == "MASE":
+            loss = self.mase_loss(
+                F, forecast, future_target, past_target, self.periodicity
+            )
+        else:
+            raise ValueError(
+                f"Invalid value {self.loss_function} for argument loss_function."
+            )
+
+        return loss
+
+
+class NBEATSPredictionNetwork(NBEATSNetwork):
+    @validated()
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    # noinspection PyMethodOverriding,PyPep8Naming
+    def hybrid_forward(
+        self, F, past_target: Tensor, future_target: Tensor = None
+    ) -> Tensor:
+        """
+
+        Parameters
+        ----------
+        F
+        past_target
+            Tensor with past observations.
+            Shape: (batch_size, context_length, target_dim).
+        future_target
+            Not used.
+
+        Returns
+        -------
+        Tensor
+            Prediction sample. Shape: (samples, batch_size, prediction_length).
```

Comment:
```suggestion
            Prediction sample. Shape: (1, batch_size, prediction_length).
```

Reply:
I see, the documentation from where I copied this seems to be out of date. The first two are switched.

Reply:
Oh I see: could you open PR to fix these doc issues as you find them? In any case, if `samples` is `1` in this case, it's better to put `1` directly, whatever the axis

Reply:
Yeah I changed it to 1. Somehow I missed to update the doc here. I'll open a PR for this.

Reply:
```        
        Tensor
            Prediction sample. Shape: (samples, batch_size, prediction_length).
        """
        distr = self.get_distr(F, past_target)

        # (num_samples, batch_size, prediction_length)
        samples = distr.sample(self.num_parallel_samples)

        # (batch_size, num_samples, prediction_length)
        return samples.swapaxes(0, 1)```

This is from simple feed forward. Interestingly the comment is correct, but the doc isnt.